x-pack/plugins/apm/public/components/app/service_map/Popover/index.tsx

```diff
@@ -14,13 +22,39 @@ import React, {
   useRef,
   useState,
 } from 'react';
-import { EuiPopover } from '@elastic/eui';
-import cytoscape from 'cytoscape';
+import {
+  SERVICE_NAME,
+  SPAN_TYPE,
+} from '../../../../../common/elasticsearch_fieldnames';
 import { useTheme } from '../../../../hooks/use_theme';
-import { SERVICE_NAME } from '../../../../../common/elasticsearch_fieldnames';
 import { CytoscapeContext } from '../Cytoscape';
-import { getAnimationOptions } from '../cytoscape_options';
-import { Contents } from './Contents';
+import { getAnimationOptions, popoverWidth } from '../cytoscape_options';
+import { BackendContents } from './backend_contents';
+import { ExternalsListContents } from './externals_list_contents';
+import { ResourceContents } from './resource_contents';
+import { ServiceContents } from './service_contents';
+
+function getContentsComponent(selectedNodeData: cytoscape.NodeDataDefinition) {
+  if (
+    selectedNodeData.groupedConnections &&
+    Array.isArray(selectedNodeData.groupedConnections)
+  ) {
+    return ExternalsListContents;
+  }
+  if (selectedNodeData[SERVICE_NAME]) {
+    return ServiceContents;
+  }
+  if (selectedNodeData[SPAN_TYPE] === 'resource') {
+    return ResourceContents;
+  }
```

Comment:
Is this the RUM agent(s) fall-back? Should we exclude by agent rather than by span type?

Reply:
If it's an HTTP request (external/http) it could be called by a RUM agent or any other service, so we could be excluding links that do have backend data.

I suppose we could also make it so these do have stats and links, but then we would have the asymmetry of them being excluded from the dependencies table for RUM services.

Reply:
not sure I follow, we exclude all RUM calls, regardless of span types in the backend views

Reply:
The node data on the client does not have the agent name of the service it was called by, so we would have to make some modifications on the server to get this.

The node data available looks like:

```js
{
  id: ">sqlite"
  label: "sqlite"
  span.destination.service.resource: "sqlite"
  span.subtype: "sqlite"
  span.type: "db"
}
```

So I think looking for `resource` is a reasonable way to do this for now.